Sources/TokamakStaticHTML/Shapes/_ShapeView.swift

```diff
@@ -56,26 +56,78 @@ extension _ShapeView: _HTMLPrimitive {
   @_spi(TokamakStaticHTML)
   public var renderedBody: AnyView {
     let path = shape.path(in: .zero).renderedBody
-    let attributes: [HTMLAttribute: String]
+    var attributes: [HTMLAttribute: String] = [:]
+    var svgDefs: AnyView?
 
     if let shapeAttributes = shape as? ShapeAttributes {
       attributes = shapeAttributes.attributes(style)
-    } else if let color = style.resolve(
-      for: .resolveStyle(levels: 0..<1),
-      in: environment,
-      role: Content.role
-    )?.color(at: 0) {
-      attributes = ["style": "fill: \(color.cssValue(environment));"]
-    } else if let foregroundStyle = environment._foregroundStyle,
-              let color = foregroundStyle.resolve(
-                for: .resolveStyle(levels: 0..<1),
-                in: environment,
-                role: Content.role
-              )?.color(at: 0)
-    {
-      attributes = ["style": "fill: \(color.cssValue(environment));"]
     } else {
-      return path
+      let resolved = style.resolve(
+        for: .resolveStyle(levels: 0..<1),
+        in: environment,
+        role: Content.role
+      )
+      switch resolved {
+      case let .gradient(gradient, style):
+        let stops = ForEach(Array(gradient.stops.enumerated()), id: \.offset) {
+          HTML("stop", [
+            "offset": "\($0.element.location * 100)%",
+            "stop-color": $0.element.color.cssValue(environment),
+          ])
+        }
+        let id = Int.random(in: 0..<Int.max)
+        attributes = ["style": "fill: url(#gradient\(id));"]
+        switch style {
```

Comment:
Looks like only linear and radial gradients are implemented here. Are you not planning to implement the others for this PR?

When you add the others, I'd suggest refactoring this `switch` to reduce code duplication.

Reply:
I don't think SVG supports the other gradient types, but I could look for a workaround.

Reply:
CSS gradients look like they would work:

https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/conic-gradient()

Reply:
Can you use CSS gradient in SVG?

Reply:
You can put the SVG in a clipPath, then use a foreignObject containing the CSS gradient with the clipPath.

See [here](https://css-tricks.com/my-struggle-to-use-and-animate-a-conic-gradient-in-svg/) for an example.

Would this approach work for our purposes?

Reply:
I'd really prefer other gradient types to be split into separate PRs. Ideally we'd want to keep PRs as small as possible, which makes it easier to review them and to go through history if needed.